tools/tcat_ble_client/GENERATING_CERTIFICATES.md

```diff
@@ -0,0 +1,93 @@
+# BBTC X.509 certificates generation
+
+---
+
+# Overview
+
+TCAT uses X.509 Certificate Extensions to provide permissions with certificates.
+
+# Extensions
+
+Extensions were introduced in version 3 of the X.509 standard for certificates. They allow certificates to be customised to applications by supporting the addition of arbitrary fields in the certificate. Each extension, identified by its OID (Object Identifier), is marked as "Critical" or "Non-Critical", and includes the extension-specific data.
+
+# Certificates generation
+
+Thread uses Elliptic Curve Cryptography (ECC), so we use `ecparam` `openssl` argument to generate the keys.
+
+## Root certificate
+
+- Generate the private key:
+
+```
+openssl ecparam -genkey -name prime256v1 -out ca_key.pem
+```
+
+- We can then generate the **.csr** (certificate signing request) file, which will contain all the parameters of our final certificate:
+
+```
+openssl req -new -sha256 -key ca_key.pem -out ca.csr
+```
+
+- Finally, we can generate the certificate itself:
+
+```
+openssl req -x509 -sha256 -days 365 -key ca_key.pem -in ca.csr -out ca_cert.pem
+```
+
+- See the generated certificate using
+
+```
+openssl x509 -in ca_cert.pem -text -noout
+```
```

Comment:
```suggestion
1. Generate the private key:

   ```
   openssl ecparam -genkey -name prime256v1 -out ca_key.pem
   ```

1. Then generate the **.csr** (certificate signing request) file, which will contain all the parameters of our final certificate:

   ```
   openssl req -new -sha256 -key ca_key.pem -out ca.csr
   ```

1. Finally, generate the certificate itself:

   ```
   openssl req -x509 -sha256 -days 365 -key ca_key.pem -in ca.csr -out ca_cert.pem
   ```

1. See the generated certificate using:

   ```
   openssl x509 -in ca_cert.pem -text -noout
   ```
```

Reply:
The preview doesn't look right, but it should format the list as a numbered step list with the code examples indented for each bullet.